### PR TITLE
12.0.3

### PR DESCRIPTION
**Pull request by kay0u**
## Solution
- *New version*
- *Restart and reduce waiting time for gitlab services*

## PR Status
- [x] Code finished.
- [x] Tested with Package_check.
- [x] Fix or enhancement tested.
- [x] Upgrade from last version tested.
- [x] Can be reviewed and tested.

## Validation
---
- [x] **Code review** : Maniack C
- [x] **Approval (LGTM)** :  Maniack C
*Code review and approval have to be from a member of @YunoHost-Apps/apps-group*
- **CI succeeded** : 
[Build Status/badge/icon)](https://ci-apps-dev.yunohost.org/jenkins/job/gitlab_ynh%20(Kay0u)/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by kay0u** [comment]
Package_checked, everything is working!

**Comment by maniackcrudelis** [comment]
The CI failed because its hard drive is too small for gitlab...
Gitlab needs almost 2Gb after being setup.

I'll have a look later...

**Comment by kay0u** [comment]
12.0.1 is out, 12.0.2 is coming

**Comment by kay0u** [comment]
Updated to 12.0.2

**Comment by maniackcrudelis** [comment]
Unfortunately, there's no way to extend the storage of a VPS on scaleway... Unless to have a much expensive VPS...
So, we'll have to make with 20Gb only...
Is there's any way on gitlab side to use less space ?

**Comment by kay0u** [comment]
No... This thing is big, and there's nothing we can do to reduce its size. /o\

I can run it on the ci-apps-dev if needed.

**Comment by maniackcrudelis** [comment]
Ok then, for now I think that the best thing to do.

**Comment by kay0u** [comment]
Error message:

> gitlab preinstall: It seems you are upgrading from 11.x version series
gitlab preinstall: to 12.x series. It is recommended to upgrade
gitlab preinstall: to the last minor version in a major version series first before
gitlab preinstall: jumping to the next major version.
gitlab preinstall: Please follow the upgrade documentation at https://docs.gitlab.com/ee/policy/maintenance.html#upgrade-recommendations
gitlab preinstall: and upgrade to 11.11 first.

I think I have to do something like nextcloud --'

**Comment by kay0u** [comment]
\o/ package_checked! It's working.

==> Ready for review!

File: scripts/upgrade.d/upgrade.11.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+gitlab_arm_source_sha256="366e12b1f3d3b1694fcb6f13da9de908360ba93f75768d97e8d01e61e8652705"
+
+gitlab_filename="gitlab-ce-${gitlab_version}.deb"
+
+# Action to do in case of failure of the package_check
+package_check_action() {
+        ynh_replace_string --match_string="command \"cat \/etc\/sysctl.conf \/etc\/sysctl.d\/\*.conf  | sysctl -e -p -\"" --replace_string="command \"cat \/etc\/sysctl.conf\"" --target_file="$final_path/embedded/cookbooks/package/resources/sysctl.rb"
+}
+
```
Comment (maniackcrudelis) [inline scripts/upgrade.d/upgrade.11.sh]: Is it really useful to declare this function here if you have to declare it again in each upgrade script ?
It seems easier to just use that line in the install script and the upgrade script.

Comment (kay0u) [reply scripts/upgrade.d/upgrade.11.sh]: It's not the same action for the version 11.x and 12.x:
```bash
 		ynh_replace_string --match_string="command \"sysctl -e --system\"" --replace_string="command \"sysctl -e --system || true\"" --target_file="$final_path/embedded/cookbooks/package/resources/sysctl.rb"
```

So during the upgrade I use the good one for the upgrade I do.

Comment (maniackcrudelis) [reply scripts/upgrade.d/upgrade.11.sh]: Indeed, haven't read the code after `ynh_replace_string`.

**Comment by maniackcrudelis** [comment]
Can be merged in 3 days.

**Pull request merged**